src/core/border_router/routing_manager.cpp

```diff
@@ -2139,9 +2141,17 @@ void RoutingManager::LocalOnLinkPrefix::Start(void)
 
 void RoutingManager::LocalOnLinkPrefix::Stop(void)
 {
+    mTimer.Stop();
     VerifyOrExit(mState != kIdle);
```

Comment:
Is it possible that `mState` is `kIdle` but it's still deprecating `mOldPrefix`?
Maybe we still need to process mOldPrefix when `mState == kIdle` ?

Reply:
I agree. I think it may be possible but unlikely (we should be advertising the local on-prefix and then have xpanid change for us to remember the old prefix to deprecate it).

I will move the `if (mOldprefix.GetLength() != 0)` up before the `VerifyOrExit()`.

Reply:
Seems managing the current prefix, and deprecating the previous prefix are indeed to different issues. 

Moreover, what happens if extpanid changes frequently. I guess we can only deprecate the previous one and stop deprecating even older prefixes?

Reply:
Yes. They are two (sort of) independent  processes. The check for `old prefix` is now before the state check (in new push).

Yes. We remember and deprecate only one old prefix (we can extend it, but I think we need to limit it and one seems to be reasonable value for now (assuming frequent xpanid change would not be a typical use-case).